### ISSUE TITLE
Dry run (-n) error if the builddir directory does not exist in 1.12.0

### ISSUE THREAD

**Issue by sean-eveson-sony** [ninja-build/ninja#2431]
Hi. I'm using Ninja 1.12.0 on Windows.

Since 1.12.0 dry run commands will produce an error such as `error: WriteFile(ninja_builddir/.ninja_lock): Unable to create file. No such file or directory` if the builddir directory does not already exist. I reproduced the error with a simple build.ninja file and running `ninja -n`:

```
builddir = ninja_builddir

rule cc
    command = clang -c $in -o $out

build test.obj: cc test.cpp
```

Thanks!

**Comment by jhasse** [comment]
I think we should fix this regression and then release 1.12.1. Help wanted :)

**Comment by jhasse** [comment]
@jdrouhard can you have a look?
